pages/_app.js

```diff
@@ -1,31 +1,47 @@
-import { useState } from "react";
+import { useEffect, useState } from "react";
 import GlobalStyle from "../styles";
 import { dummyData } from "./api/dummyData";
 import Layout from "@/components/Layout";
 import * as wanakana from "wanakana";
 import { convertToKana } from "@/utils/helperFunctions";
+import refactorDictionaryOutput from "@/utils/refactorDictionaryOutput";
+import useSWR from "swr";
 
+const fetcher = async (url) => {
+  const response = await fetch(url);
+  if (!response.ok) {
+    const error = new Error("An error occurred while fetching the data.");
+
+    error.info = await response.json();
+    error.status = response.status;
+    throw error;
+  }
+
+  return response.json();
```

Comment:
```suggestion
  return await response.json();
```
Frankly, I'm not sure if that makes a difference if you leave out the `await`. You may test it yourself, with and without `await`.